sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,1299 @@
+                    };
+                };
+                readonly Faxes: {
+                    readonly type: "array";
+                    readonly items: {
+                        readonly type: "phoneNumber";
+                        readonly description: "Fax number";
+                        readonly example: "+1 (987) 312-6745";
+                    };
+                };
+                readonly OtherPhones: {
+                    readonly type: "array";
+                    readonly items: {
+                        readonly type: "phoneNumber";
+                        readonly description: "Other phone number";
+                        readonly example: "+1 (987) 213-5673";
+                    };
+                };
+                readonly Emails: {
+                    readonly type: "array";
+                    readonly items: {
+                        readonly type: "string";
+                        readonly description: "Contact email";
+                        readonly example: "chris.smith@contoso.com";
+                    };
+                };
+                readonly Websites: {
+                    readonly type: "array";
+                    readonly items: {
+                        readonly type: "string";
+                        readonly description: "Website";
+                        readonly example: "https://www.contoso.com";
+                    };
+                };
+            };
+        };
+    };
+};
+
+// @public
+export interface CommonClientOptions extends PipelineOptions {
+    apiVersion?: FormRecognizerApiVersion;
+    stringIndexType?: StringIndexType;
+}
+
+// @public
+export type ContentType = "application/octet-stream" | "application/pdf" | "image/bmp" | "image/jpeg" | "image/png" | "image/tiff";
+
+// @public
+export interface CopyAuthorization {
+    accessToken: string;
+    expirationDateTime: Date;
+    targetModelId: string;
+    targetModelLocation: string;
+    targetResourceId: string;
+    targetResourceRegion: string;
+}
+
+// @public
+export interface CopyModelOptions extends OperationOptions, PollerOptions<TrainingPollOperationState> {
+}
+
+// @public
+export interface CustomDocumentModelsInfo {
+    count: number;
+    limit: number;
+}
+
+// @public
+export interface DateFieldSchema {
+    // (undocumented)
+    readonly type: "date";
+}
+
+// @public
+export interface DeleteModelOptions extends OperationOptions {
+}
+
+// @public
+export interface DocTypeInfo {
+    description?: string;
+    fieldConfidence?: {
+        [propertyName: string]: number;
+    };
+    fieldSchema: {
+        [propertyName: string]: DocumentFieldSchema;
+    };
+}
+
+// @public
+export class DocumentAnalysisClient {
+    constructor(endpoint: string, credential: TokenCredential, options?: DocumentAnalysisClientOptions);
+    constructor(endpoint: string, credential: KeyCredential, options?: DocumentAnalysisClientOptions);
+    // @internal
```

Comment:
NIT: I do not think there is much value in creating separate overloads here.

Reply:
The third internal overload makes it so that an expression with the type `TokenCredential | KeyCredential` is assignable to the second parameter of the constructor function. Switching to the overload style actually broke some of my test code. IMO the docs experience with this form is just _slightly_ better. It makes the docstrings less lengthy and can show contextual documentation that the user can page through in IntelliSense. It's debatable, but after we talked about this a few weeks back in the team meeting I compared them and thought this way yielded slightly nicer intellisense.

Reply:
> Switching to the overload style actually broke some of my test code. 

Not sure how to interpret this. Anyway, IMO we should be consistent across packages on this so can probably have a discussion around it in the design guidelines meetings.

Reply:
Switching to an overload from a union is a breaking change unless the third internal overload is also added.

I don't believe we are consistent here. It'd be good to solidify a pattern. What does autorest do in this case?

Reply:
autorest only understands TokenCredential so it does not generate a union.